staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -239,10 +243,24 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+
+	// if the scope in the CRD and the scope in request differ (with exception of the verbs in possiblyAcrossAllNamespacesVerbs
+	// for namespaced resources), pass request to the delegate, which is supposed to lead to a 404.
+	namespacedCRD, namespacedReq := crd.Spec.Scope == apiextensions.NamespaceScoped, len(requestInfo.Namespace) > 0
```

Comment:
Not sure whether we need to add some multi-tenancy related changes. Please ask Qian to review.

Reply:
This change looks fine to me. 

We would like to keep the CRD spec backwards-compatible so all the CRD yamls can be re-used. So there will no changes to the interpretation of the CRD specs. We are not going to introduce a new scope in the CRD spec.

Reply:
This code looks fine to me in respect to multi-tenancy. "r.delegate.ServeHTTP(w, req)" is the error handler and thus does not need any multi-tenancy related change.